[DOC] Update code formatting description in `CONTRIBUTING.md` (#2233)
NetEase/amoro: CONTRIBUTING.md

# Contributing

This document provides guidelines for contributing to Amoro. While these suggestions are not strict
rules, they aim to facilitate a smooth contribution experience.

## Issues

Regardless of the type of contribution you plan to make, it is recommended that you create an issue
to track it.

* Before creating an issue, please search within the issues to see if a similar one has already been
  reported.
* Choose the appropriate type:
    * Feature: A new feature to be added.
    * Improvement: Enhancement of an existing feature, including code quality, performance, user
      experience, etc.
    * Bug: A problem that prevents the project from functioning as intended.
    * Subtask: A subtask of a Feature/Improvement that can be broken down into smaller steps.

## Pull requests

Pull requests are the preferred mechanism for contributing to Amoro

* Generally, create a PR only to the master branch.
* PR should be linked to the corresponding issue.

>     * The PR title format should be: \[ARCTIC-{issue_number}\]\[{module}\]{pr_description}.

Reviewer: ```suggestion
    * The PR title format should be: \[AMORO-{issue_number}\]\[{module}\]{pr_description}.
```

## Design document

Write down your implementation plan and discuss it with other developers in the community before you

> start coding officially. If it is just a small changes, describe the implementation steps clearly in

Reviewer 2: ```suggestion
start coding officially. If it is just a small change, describe the implementation steps clearly in
```

the Issue. If it is a relatively large work, it is recommended to write a design document for this
feature. Here is

> a [design document template](https://docs.google.com/document/d/1LeTyrlzQJfSs2DkRBsucK_vV5gtHRYLb1KSrpu0hp3g/edit?usp=sharing)

Reviewer 3: The design document template should be rename from `Arctic` to `Amoro`.

Author: Thanks for the reminder, I have changed the doc name and translated the doc to the English version.

You can format your code by executing the command `mvn spotless:apply` in the root directory of
project.

Or you can configure your IDEA to automatically format your code. Then you will need to install
the [google-java-format](https://github.com/google/google-java-format) plugin. However, a specific
version of this plugin is required.
Download [google-java-format v1.7.0.6](https://plugins.jetbrains.com/plugin/8527-google-java-format/versions/stable/115957)
and install it as follows. Make sure to never update this plugin.

1. Go to “Settings/Preferences” → “Plugins”.
2. Click the gear icon and select “Install Plugin from Disk”.
3. Navigate to the downloaded ZIP file and select it.

After installing the plugin, format your code automatically by applying the following settings:

1. Go to “Settings/Preferences” → “Other Settings” → “google-java-format Settings”.
2. Tick the checkbox to enable the plugin.
3. Change the code style to “Default Google Java style”.
4. Go to “Settings/Preferences” → Editor → Code Style → Scala.
5. Change the “Formatter” to “scalafmt”.
6. Go to “Settings/Preferences” → “Tools” → “Actions on Save”.

> 7. Under “Formatting Actions”, select “Optimize imports” and “Reformat file”.

Reviewer 3: Should `Reformat file` be `Reformat code` ? (My IntelliJ IDEA 2021.2.3 (Community Edition))
<img width="1819" alt="image" src="https://github.com/NetEase/amoro/assets/103108928/00ae6388-52b8-4697-aac9-2e7de424254e">

8. From the “All file types list” next to “Reformat code”, select Java and Scala.

### Copyright

All files (including source code, configuration files) in the project are required to declare
CopyRight information at the top, and the project uses Apache License 2. You can configure the

> copyright information in IntelliJ IDEA with follow steps:

Reviewer 2: ```suggestion
copyright information in IntelliJ IDEA with the following steps:
```
